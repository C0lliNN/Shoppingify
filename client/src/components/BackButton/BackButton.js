import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import * as variables from '../../helpers/style-constants';

const StyledButton = styled.button`
  border: none;
  cursor: pointer;
  background: transparent;
  outline: none;
  color: ${variables.COLORS.primary};
  font-size: ${variables.FONT_SIZE_2};
  font-family: ${variables.FONT_FAMILY};
  font-weight: 600;
  display: flex;
  align-items: center;
`;
const Icon = styled.i`
  color: ${variables.COLORS.primary};
  font-size: ${variables.FONT_SIZE_3};
`;

function BackButton({ onClick }) {
  return (
    <StyledButton onClick={onClick}>
      <Icon className="material-icons-round">keyboard_backspace</Icon>
      <span>&nbsp;back</span>
    </StyledButton>
  );
}

BackButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default BackButton;
