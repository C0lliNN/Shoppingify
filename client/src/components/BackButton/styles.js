import styled from 'styled-components';
import * as variables from '../../helpers/style-constants';

export const StyledButton = styled.button`
  border: none;
  cursor: pointer;
  background: transparent;
  outline: none;
  color: ${variables.COLORS.primary};
  font-size: ${variables.FONT_SIZE_2};
  font-family: ${variables.FONT_FAMILY};
  font-weight: 600;
  display: flex;
  align-items: center;
`;
export const Icon = styled.i`
  color: ${variables.COLORS.primary};
  font-size: ${variables.FONT_SIZE_3};
`;
