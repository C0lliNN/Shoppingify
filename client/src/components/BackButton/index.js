import PropTypes from 'prop-types';
import React from 'react';
import { StyledButton, Icon } from './styles';

function BackButton({ onClick }) {
  return (
    <StyledButton onClick={onClick}>
      <Icon className="material-icons-round">keyboard_backspace</Icon>
      <span>&nbsp;back</span>
    </StyledButton>
  );
}

BackButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default BackButton;
