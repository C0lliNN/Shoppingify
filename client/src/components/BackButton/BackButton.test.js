import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import BackButton from './BackButton';
import 'jest-styled-components';

describe('<BackButton/>', () => {
  it('should render correctly', () => {
    const { container } = render(<BackButton onClick={() => {}} />);
    expect(container).toMatchSnapshot();
  });
  it('should execute the onClick function when the the button is clicked', () => {
    const fn = jest.fn();
    const { getByText } = render(<BackButton onClick={fn} />);

    const element = getByText('back');
    fireEvent.click(element);

    expect(fn).toHaveBeenCalled();
  });
});
