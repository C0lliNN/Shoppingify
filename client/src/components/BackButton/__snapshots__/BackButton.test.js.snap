// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton/> should render correctly 1`] = `
.c0 {
  border: none;
  cursor: pointer;
  background: transparent;
  outline: none;
  color: #F9A109;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: 600;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  color: #F9A109;
  font-size: 18px;
}

<div>
  <button
    class="c0"
  >
    <i
      class="c1 material-icons-round"
    >
      keyboard_backspace
    </i>
    <span>
      Â back
    </span>
  </button>
</div>
`;
